### PR TITLE
TomWeaver18 - refactor main.py main function

### PR DESCRIPTION
**Pull request by TomWeaver18**
Refactor work that was initially accepted on the internal private repo but didn't make the move to open sourcing on github.

The initial change had a conflict caused by the reformatting (to pep8 I assume) of the original file.

This has been resolved, tested and linted.

Happy reviewing

### CODE REVIEW

File: dex/tools/Main.py [hunk 1]
```diff
@@ -39,6 +39,34 @@
 from dex.utils.Version import version
 from dex.utils import WorkingDirectory
 
+def output_bug_report_message(context):
```
Comment (gregbedwell) [inline dex/tools/Main.py]: This function is only useful from this file, so please give it a single leading underscore (see http://pep8.org/#descriptive-naming-styles ).  The same goes for the other newly added functions. 

We might need to tidy up some of the other existing functions in this file to do so too, but we may as well start off doing it right from now ;)

File: dex/tools/Main.py [hunk 2]
```diff
@@ -54,6 +82,65 @@ def get_tool_names():
     ]
 
 
+def set_auto_highlights(context):
+    # Flag some strings for auto-highlighting.
```
Comment (gregbedwell) [inline dex/tools/Main.py]: Convert to docstring.

Comment (TomWeaver18) [reply dex/tools/Main.py]: nice easy fix :) thanks for the spot

**Comment by TomWeaver18** [comment]
In response to the YAPF version comment,

I've been using YAPF at home whilst off these past couple of days and I'm getting similar reformatting changes.

Theres quite a considerable change on the version I'm using in win 10 which I installed yesterday to the one I'm using at work that is possibly months out of date.

Maybe the best thing for now would be to set up a seperate branch to handle a up-to-date yapf format pull request and get that in pronto to avoid any future pull requests being polluted with a large change list (as I'm seeing now).

Cheers
Tom

File: dex/tools/Main.py (hunk 1 above)
Comment (TomWeaver18) [reply dex/tools/Main.py]: So I've added the _ to the all methods that don't then cause an issue with anything else in the project. 

the __init__.py file has several references to functions in Main.py, changing the functions to the internal style causes issues, so I've left those alone.

Cheers
Tom

Comment (gregbedwell) [reply dex/tools/Main.py]: Great.  Thanks!
The _ prefix is akin to the C static keyword, or a C++ anonymous namespace, so it makes sense not to use it for anything that we're actually exporting.

**Comment by TomWeaver18** [comment]
re-worked the patch to coincide with Gregg's feedback.

Also a quick little stealth change, removed some on the new line spacings in the tool_main function as it wasn't really aiding readability (imo).

cheers
tom

**Comment by TomWeaver18** [comment]
I'd really like to get this patch landed if possible as I'm now making more changes in this area.

any chance we can get this merged in?

**Comment by TomWeaver18** [comment]
ping

**Pull request merged**